crates/db_views_actor/src/community_follower_view.rs

```diff
@@ -1,21 +1,42 @@
 use crate::structs::CommunityFollowerView;
+use chrono::Utc;
 use diesel::{
   dsl::{count_star, not},
   result::Error,
-  sql_function,
   ExpressionMethods,
   QueryDsl,
 };
 use diesel_async::RunQueryDsl;
 use lemmy_db_schema::{
-  newtypes::{CommunityId, DbUrl, PersonId},
+  newtypes::{CommunityId, DbUrl, InstanceId, PersonId},
   schema::{community, community_follower, person},
-  utils::{get_conn, DbPool},
+  utils::{functions::coalesce, get_conn, DbPool},
 };
 
-sql_function!(fn coalesce(x: diesel::sql_types::Nullable<diesel::sql_types::Text>, y: diesel::sql_types::Text) -> diesel::sql_types::Text);
-
 impl CommunityFollowerView {
+  /// return a list of local community ids and remote inboxes that at least one user of the given instance has followed
+  pub async fn get_instance_followed_community_inboxes(
+    pool: &mut DbPool<'_>,
+    instance_id: InstanceId,
+    published_since: chrono::DateTime<Utc>,
+  ) -> Result<Vec<(CommunityId, DbUrl)>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    // todo: in most cases this will fetch the same url many times (the shared inbox url)
```

Comment:
This should be prevented by `distinct()`, no?

Reply:
Yes - PG will only send a single copy to rust, but it has to scan through all follower rows (same as it was before). So on the PG side it would be possible to optimize this further by adding e.g. a new table community_followed_instances (community_id, instance_id) that would work for all instances that support fully shared inboxes. It would be a bit more complicated though to keep it in sync.

I'll update the comment in the code base to be clearer.

Reply:
> e.g. a new table community_followed_instances (community_id, instance_id) that would work for all instances that support fully shared inboxes.

Incidentally, this is possibly a recommendation I was going to make for a federation issue/behavior I discovered a few minutes in testing Lemmy for the first time.

Unfortunately, I still need some free time to complete my investigation, write a proper RFC, and maybe write some tests and possible solutions too...

Keeping it in sync will not be an issue, but explaining why will spill the beans on my potential future RFC ;)

But still, with the current code, would checking this table with each follow/unfollow be really a performance issue? My guess was that it wouldn't be much of an issue because: a) The following/unfollowing is not that frequent of an action and b) because the performance cost should be minimal anyway!